src/marks/difference.js

```diff
@@ -0,0 +1,109 @@
+import {area as shapeArea, line as shapeLine} from "d3";
+import {create} from "../context.js";
+import {maybeCurve} from "../curve.js";
+import {Mark} from "../mark.js";
+import {identity, indexOf, isColor} from "../options.js";
+import {applyIndirectStyles, applyTransform, getClipId} from "../style.js";
+import {maybeDenseIntervalX} from "../transforms/bin.js";
+
+const defaults = {
+  ariaLabel: "difference",
+  fill: "none",
+  stroke: "currentColor",
+  strokeWidth: 1.5,
+  strokeLinecap: "round",
+  strokeLinejoin: "round",
+  strokeMiterlimit: 1
+};
+
+function maybeColor(value) {
+  if (value == null) return "none";
+  if (!isColor(value)) throw new Error(`invalid color: ${value}`);
+  return value;
+}
+
+class Difference extends Mark {
+  constructor(data, options = {}) {
+    const {x1, y1, x2, y2, curve, tension, positiveColor = "green", negativeColor = "blue"} = options;
```

Comment:
these default colors are not very pretty; maybe use blue/orange instead?

Reply:
I’ll pick some prettier colors.